crates/miden-tx/src/executor/notes_checker.rs

```diff
@@ -0,0 +1,83 @@
+use alloc::vec::Vec;
+
+use miden_lib::{
+    account::interface::NoteAccountCompatibility, note::well_known_note::WellKnownNote,
+};
+use miden_objects::{
+    account::AccountId,
+    block::BlockNumber,
+    note::NoteId,
+    transaction::{InputNote, TransactionArgs},
+};
+use winter_maybe_async::{maybe_async, maybe_await};
+
+use super::{ExecutionCheckResult, TransactionExecutor};
+use crate::TransactionExecutorError;
+
+pub struct NotesChecker {
```

Comment:
I think we should add docs to this to explain what this can be used for on a high level.